orchestrator/domain/base.py

```diff
@@ -406,12 +396,20 @@ def _data_from_lifecycle(cls, other: "DomainModel", status: SubscriptionLifecycl
                 if is_union_type(field_type):
                     if is_optional_type(field_type):
                         data[field_name] = None
-                    field_types = get_args(field_type)
-                    for f_type in field_types:
-                        if value and not isinstance(None, f_type) and f_type.name == value.name:
+                    for f_type in filter_nonetype(args):
+                        if f_type.name == value.name:
```

Comment:
can't you also use a filter for this and a if statement for the logging when field is none. instead of a for else

Reply:
Resolving for now, we can look into this later